src/core/lib/gpr/time_posix.cc

```diff
@@ -114,7 +114,9 @@ static gpr_timespec now_impl(gpr_clock_type clock) {
       break;
     case GPR_CLOCK_MONOTONIC:
       now_dbl = ((double)(mach_absolute_time() - g_time_start)) * g_time_scale;
-      now.tv_sec = (int64_t)(now_dbl * 1e-9);
+      // Add 5 seconds arbitrarily: avoids weird conditions in gprpp/time.cc
+      // when there's a small number of seconds returned.
+      now.tv_sec = (int64_t)(now_dbl * 1e-9 + 5.0);
```

Comment:
This seems very strange.  Why do we need this, and how is it safe?

Reply:
We've got some issues in gprpp/time.cc initialization about wanting values > 1 second for gpr_now(GPR_CLOCK_MONOTONIC)... and given it needs to be *relatively* fast path here, and the monotonic clock is a number of seconds from an *arbitrary* epoch, it seems just moving the epoch here to sidestep the issue during initialization is fair game.

Reply:
(this need goes away when we increase the precision of `TimeStamp`, but until then we need to take steps to ensure that we don't accidentally produce timestamps way out towards the edges of its range)